allow disassembly of blanks

misterprimus opened the pull request:
I noticed 22 blanks cannot be uncrafted into valuable gunpowder. This should fix it.

<!-- HOW TO USE: Under each "#### Heading" below, enter information relevant to your pull request.
Leave the headings unless they don't apply to your PR.

Please read carefully and don't delete the comments delimited by "< !--" and "-- >"
Once a pull request is submitted, automatic stylistic and consistency checks will be performed on the PR's changes.
The results of these can be either seen under the "Files changed" section of a PR or in the check's details.

NOTE: Please grant permission for repository maintainers to edit your PR.  It is EXTREMELY common for PRs to be held up due to trivial changes being requested and the author being unavailable to make them. -->

#### Summary
Content "allow disassembly of blanks"
<!-- This section should consist of exactly one line, edit the one above.
1. Replace the word "Category" with one of these words: Features, Content, Interface, Mods, Balance, Bugfixes, Performance, Infrastructure, Build, I18N.
2. Replace the text inside the quotes with a brief description of your changes.
Or if you don't want a changelog entry, replace the whole line with just the word "None" (with no quotes).
For more on the meaning of each category, see:
https://github.com/CleverRaven/Cataclysm-DDA/blob/master/doc/CHANGELOG_GUIDELINES.md
If approved and merged, your summary will be added to the project changelog:
https://github.com/CleverRaven/Cataclysm-DDA/blob/master/data/changelog.txt -->

#### Purpose of change

Allow disassembly of blanks to recover gunpowder and casings.

#### Describe the solution

Fixes what seems to be an oversight that excluded uncraft recipe for this item.

#### Describe alternatives you've considered

None seemed necessary.

#### Testing

Not tested.

#### Additional context

If someone thinks we need to add more uncraft recipes to some other blanks I am unaware of, perhaps we can do it with this PR since it would basically be fixing the same issue. **Also, just as heavier slugged versions of the same caliber come with less gunpowder (to prevent overpressure in the barrel), it seems some blanks are loaded with more gunpowder than ordinary rounds. Not sure if this should be reflected with _more_ gunpowder being generated from the disassembly of one of these.**

misterprimus committed:
allow disassembly of blanks

I noticed 22 blanks cannot be uncrafted into valuable gunpowder. This should fix it.

tenmillimaster commented:
This should be using a tool other than a bullet puller to open the star crimp and trim the case down. There is no bullet to pull.

A knife or screwdriver, though if these were used you wouldn't be left with a usable case. Ideally a case trimmer tool, *maybe* a pipe cutter.

misterprimus committed:
change quality requirement id PULL to SCREW_FINE

Tonkatsu made a good point re: blanks not having slugs attached and bullet pullers being less relevant here. I looked at some pictures of blanks and it seems they are crimped at the ends. A precision-tipped screwdriver can probably open one of these to release the powder.

misterprimus commented:
> This should be using a tool other than a bullet puller to open the star crimp and trim the case down. There is no bullet to pull.
> 
> A knife or screwdriver, though if these were used you wouldn't be left with a usable case. Ideally a case trimmer tool, _maybe_ a pipe cutter.

I did fine screwdriving after looking at pictures of crimped blanks. That's what I'd use based on my observations.

tenmillimaster commented:
> > This should be using a tool other than a bullet puller to open the star crimp and trim the case down. There is no bullet to pull.
> > A knife or screwdriver, though if these were used you wouldn't be left with a usable case. Ideally a case trimmer tool, _maybe_ a pipe cutter.
> 
> I did fine screwdriving after looking at pictures of crimped blanks. That's what I'd use based on my observations.

There is still the issue that the casing produced is not usable after this - .22 Powderblanks have a bottle neck that would need to be somewhat significantly resized (with a mandrel. Fireforming is out since .22LR is basically un-reprimable.). 

The simplest solution is just to make this uncraft lose the casing, as I suggested.

misterprimus committed:
modify brass generated by uncrafting blanks

misterprimus commented:
> > > This should be using a tool other than a bullet puller to open the star crimp and trim the case down. There is no bullet to pull.
> > > A knife or screwdriver, though if these were used you wouldn't be left with a usable case. Ideally a case trimmer tool, _maybe_ a pipe cutter.
> > 
> > 
> > I did fine screwdriving after looking at pictures of crimped blanks. That's what I'd use based on my observations.
> 
> There is still the issue that the casing produced is not usable after this - .22 Powderblanks have a bottle neck that would need to be somewhat significantly resized (with a mandrel. Fireforming is out since .22LR is basically un-reprimable.).
> 
> The simplest solution is just to make this uncraft lose the casing, as I suggested.
> 
> ![image](https://user-images.githubusercontent.com/7764202/281918359-836653fa-c540-4cb2-9b62-01b915c68bb7.png)

What do you think of this? It generates a different resulting brass component, which presumably would not be usable in crafting firearm-useable slugged .22 rounds.

misterprimus committed:
Update casing.json

tenmillimaster commented on data/json/items/generic/casing.json:
```diff
@@ -44,6 +44,22 @@
+    "name": { "str": "unused .22 blank casing" },
+    "symbol": "=",
+    "color": "yellow",
+    "description": "An unfired, like-new .22 round casing, with the primer still intact. This one appears to be sourced from a blank round and is thus shorter.",
```
```suggestion
    "description": "An unfired, like-new .22 short casing, with the priming compound still intact.",
```

If you're a little less specific about the casing this can also be the casing for .22 "CB" and a weird subsonic loading of .22 LR.
![image](https://github.com/CleverRaven/Cataclysm-DDA/assets/7764202/2e8f7d89-fe0c-47ad-96a0-10d853ff2733)

tenmillimaster commented on data/json/items/generic/casing.json:
```diff
@@ -44,6 +44,22 @@
+    "category": "spare_parts",
+    "price": 90,
+    "price_postapoc": 0,
+    "name": { "str": "unused .22 blank casing" },
```
```suggestion
    "name": { "str": "unused .22 short casing" },
```

misterprimus committed:
Update data/json/items/generic/casing.json

Co-authored-by: Tonkatsu <7764202+tenmillimaster@users.noreply.github.com>

misterprimus committed:
Update data/json/items/generic/casing.json

Co-authored-by: Tonkatsu <7764202+tenmillimaster@users.noreply.github.com>

misterprimus commented on data/json/uncraft/ammo/22.json:
```diff
@@ -1,4 +1,14 @@
+    "difficulty": 5,
+    "time": "5 s",
+    "qualities": [ { "id": "SCREW_FINE", "level": 1 } ],
+    "components": [ [ [ "22_casing_new", 1 ] ], [ [ "gunpowder_pistol", 1 ] ] ]
```
Can you explain this one? I applied your other two changes, but then this change became "outdated." It seems to not make sense given you added descriptions to the 22_casing_new_short and this suggestion deletes it? I assumed your latter two suggestions were the ones you wanted so merged those first, but obviously like I said this one is no longer usable.

tenmillimaster replied:
It became outdated when you changed the file, chaning the casing produced. I've resolved it as it's no longer relevant.
